config/routes.rb

```diff
@@ -2,13 +2,15 @@
   devise_for :users
   # Define your application routes per the DSL in https://guides.rubyonrails.org/routing.html
   get 'users/:user_id/articles/myarticles', to: 'articles#show_my_articles', as: 'myarticles'
+  
   resources :users do
     resources :articles do
       resources :comments
     end
   end
   # Defines the root path route ("/")
   root "articles#index"
+  get 'users/:user_id/articles/query/:args', to: 'articles#index', as:"queryarticles"
```

Comment:
why the path is `users/:user_id/articles/query/:args` ? By default route for `index` is `/articles` . Also, `:args` are handled as query params 
always use snake_case in ruby "query_articles"

Reply:
Do you know what happens when we create `resources` ? If yes, please check command `rails routes` and see how many routes you have for `articles`